string hehavior_name
int8 current_phase